app/controllers/users_controller.rb

```diff
@@ -0,0 +1,68 @@
+class UsersController < ApplicationController
+  load_and_authorize_resource
+
+  def new
+    if current_user
+      redirect_to user_path(current_user)
+    else
+      @user = User.new
+      @address = Address.new
+    end
+  end
+
+  def create
+    @user = User.new(user_params)
+    @address = Address.new(address_params)
+    @user.address = @address
+    if @user.save
+      flash[:success] = 'The user has been created!'
+      redirect_to login_path
+    else
+      flash[:danger] = "The user couldn't be updated!"
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.address.update_attributes(address_params)
+    if @user.update_attributes(user_params)
```

Comment:
look at how you updated the shop. it's better there
REMEMBER WHY I ASKED YOU TO PUT THOSE IN A TRANSACTION? it's the same issue here